pkg/devfile/parser/parse.go

```diff
@@ -212,11 +250,36 @@ func parseFromURI(uri string, curDevfileCtx devfileCtx.DevfileCtx) (DevfileObj,
 		d.Ctx = devfileCtx.NewURLDevfileCtx(u.String())
 	}
 	d.Ctx.SetURIMap(curDevfileCtx.GetURIMap())
-	// Fill the fields of DevfileCtx struct
-	err = d.Ctx.PopulateFromURL()
-	if err != nil {
-		return DevfileObj{}, err
+	return populateAndParseDevfile(d, true)
+}
+
+func parseFromRegistry(registryId, registryURL string, curDevfileCtx devfileCtx.DevfileCtx) (DevfileObj, error) {
```

Comment:
I got your point. but I think registry is much more clear than id?
even I think parent.Id is not very clear in devfile without checking the api reference

Reply:
My explanation:
- I think we are parsing from the id(which happens to be parent) and we fetch from the registry.. Hence the suggestion `parseFromId()`
- `parent.Id`/`parentId` acc to the schema definition is `Id in a registry that contains a Devfile yaml file`, its not really the registry id. Registry id would be defined in the registry's metadata perhaps. For example, registry id is [odo-devfiles](https://github.com/odo-devfiles/registry/).. Hence the suggestion `parentId`

Reply:
Here is a screenshot of how devfile v1 used id for plugin:
![Screen Shot 2021-03-12 at 2 29 26 PM](https://user-images.githubusercontent.com/31771087/110989155-77bd3900-833f-11eb-9cfa-9e5bb4f50abf.png)

I think `che-incubator` is the registryId here and I think this may also tie back to this discussion https://github.com/devfile/library/pull/63#discussion_r593361613

Reply:
For v2, I have asked Jingfu and confirmed with Elson in our meeting; all valid registries should follow the rule defined in: https://github.com/devfile/api/blob/master/docs/proposals/registry/index-server-REST-API.md
To get the devfile content, the request should be sent to `http://{registry host}/devfiles/{stack}`
Here `http://{registry host}` or `{registry host}` is the `registryUrl`, and `{stack}` is the `Id` defined in devfile.